### PR TITLE
Go toolchain contains all packages

### PR DESCRIPTION
**Pull request by ctwarog-ross**
## 📖 Description
Closes Issue#1303

All package installs moved to toolchain image
Packages version-locked in toolchain.env
Affected protos re-compiled

---

## ✅ Definition of Done (DoD)
- [x] Feature is implemented and works as expected
- [x] Edge cases are handled
- [x] Tests are added or updated (if applicable)
- [ ] Documentation is updated (if needed)
- [ ] Code is reviewed and approved 

---

## 🛠️ Implementation Notes

---

## 🔗 Related Issues / Links
[
](https://github.com/rossvideo/Catena/issues/1303)

---
## 📝 Additional Context

---

## 🚧 Risks / Open Questions

---
## ✨ Updates
<!-- (Optional) List summarizing changes made when editing the pull request>

### CODE REVIEW

**Comment by ctwarog-ross** [comment]
bugbot run

**Comment by ctwarog-ross** [comment]
bugbot run

**Comment by ctwarog-ross** [comment]
bugbot run

File: .devcontainer/go/devcontainer.dockerfile [hunk 1]
```diff
@@ -40,20 +40,10 @@ RUN groupadd -g $USER_GID $USER_NAME && \
...
 
-RUN go install golang.org/x/tools/gopls@latest \
-    && go install github.com/go-delve/delve/cmd/dlv@latest \
-    && go install google.golang.org/protobuf/cmd/protoc-gen-go@latest \
-    && go install google.golang.org/grpc/cmd/protoc-gen-go-grpc@latest \
-    && go install github.com/axw/gocov/gocov@latest \
-    && go install github.com/matm/gocov-html/cmd/gocov-html@latest \
-    && go install github.com/jandelgado/gcov2lcov@latest
-    
 RUN export PATH="$PATH:$(go env GOPATH)/bin"
```
Comment (abrown-RV) [inline .devcontainer/go/devcontainer.dockerfile]: don't really need this anymore

File: .devcontainer/go/toolchain.dockerfile
Comment (abrown-RV) [inline .devcontainer/go/toolchain.dockerfile]: obsolete as well

File: .devcontainer/go/toolchain.dockerfile [hunk 2]
```diff
@@ -56,5 +56,23 @@ COPY smpte/install-tooling.sh /root/smpte/install-tooling.sh
...
 
+# Install protobuf compiler and Go tools
+RUN . /root/toolchain.env \
+    && apt-get update \
+    && apt-get install --no-install-recommends -y protobuf-compiler=${PROTOBUF_COMPILER_VERSION} \
+    && apt-get clean
+
+ENV GOBIN=/usr/local/bin
+
+RUN go install golang.org/x/tools/gopls@v0.21.1 \
```
Comment (abrown-RV) [inline .devcontainer/go/toolchain.dockerfile]: so there's a fancy go way to pin these versions basically you turn the devcontainer/go dir into another go module. Inside that dir you run `go get -tool <the tool>` and it will build up a normal go.mod file. Then in the dockerfile you do
```dockerfile
WORKDIR /toolchain

ENV GOBIN=/usr/local/bin

COPY .devcontainer/go/go.mod .devcontainer/go/go.sum ./

RUN go mod download

RUN go install \
    <the same list WITHOUT VERSIONS>
```
so the go mod download grabs the versions and then go install uses those pinned versions

File: .github/workflows/go-pull-request.yml
Comment (abrown-RV) [inline .github/workflows/go-pull-request.yml]: no longer installing, can we be not root?

**Review by abrown-RV — CHANGES_REQUESTED** [review]

File: .devcontainer/go/go.mod [hunk 3]
```diff
@@ -0,0 +1,62 @@
+module catena-go-tools
```
[the remark this answers is not in the document]
Comment (ctwarog-ross) [reply .devcontainer/go/go.mod:1]: good trivy

**Comment by ctwarog-ross** [comment]
bugbot run

**Comment by ctwarog-ross** [comment]
bugbot run

File: .github/workflows/go-pull-request.yml [hunk 4]
```diff
@@ -164,35 +142,17 @@ jobs:
...
-          # Create non-root user for testing
-          useradd -m -s /bin/bash testuser 2>/dev/null || true
-          
-          # Ensure workspace is owned by testuser
-          chown -R testuser:testuser ${GITHUB_WORKSPACE}
-          
-          # Run tests as non-root to properly test permission errors
-          # Using runuser to preserve environment including PATH
-          runuser -u testuser -- bash -c "cd ${GITHUB_WORKSPACE}/sdks/go && make COVERAGE_DIR=${GITHUB_WORKSPACE}/sdks/go coverage coverage-html"
+          cd ${GITHUB_WORKSPACE}/sdks/go && make COVERAGE_DIR=${GITHUB_WORKSPACE}/sdks/go coverage coverage-html
```
Comment (abrown-RV) [inline .github/workflows/go-pull-request.yml]: split into 2 lines

**Review by abrown-RV — APPROVED** [review]

**Review by abrown-RV — CHANGES_REQUESTED** [review]
bugbot made a good point on the toolchain build, it should watch the new go mod and sum files.

**Review by abrown-RV — APPROVED** [review]
